src/compile/selection/transforms/index.ts

```diff
@@ -0,0 +1,33 @@
+import {Model} from '../../model';
+import {UnitModel} from '../../unit';
+import {SelectionSpec, SelectionComponent} from '../../../selection';
+import {Dict} from '../../../util';
+
+export interface TransformCompiler {
+  has: (sel: SelectionComponent | SelectionSpec) => boolean;
+  parse?: (model: UnitModel, def: SelectionSpec, sel: SelectionComponent) => void;
+  signals?: (model: UnitModel, sel: SelectionComponent, signals: any[]) => any[];
+  topLevelSignals?: (model: Model, sel: SelectionComponent, signals: any[]) => any[];
+  // tupleExpr?: (model: UnitModel, sel: SelectionComponent, expr: string) => string;
+  modifyExpr?: (model: UnitModel, sel: SelectionComponent, expr: string) => string;
+  marks?: (model: UnitModel, sel:SelectionComponent, marks: any[], selMarks: any[]) => any[];
+  clippedGroup?: boolean;
+}
+
+import project from './project';
+import toggle from './toggle';
+import translate from './translate';
+import zoom from './zoom';
+import scales from './scales';
+import inputs from './inputs';
+import nearest from './nearest';
+const compilers: Dict<TransformCompiler> = {project, toggle, scales,
+  translate, zoom, inputs, nearest};
+
+export function transforms(sel: SelectionComponent, cb: (tx: TransformCompiler) => void) {
```

Comment:
Could you expand on why you consider this method convoluted? It's a fairly straightforward implementation of a [visitor pattern](https://en.wikipedia.org/wiki/Visitor_pattern). I could rename it to `transformsVisitor` but I erred on the side of brevity.